fuzz_test.go

```diff
@@ -0,0 +1,90 @@
+package orderedmap
+
+// Adapted from https://github.com/dvyukov/go-fuzz-corpus/blob/c42c1b2/json/json.go
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/dvyukov/go-fuzz-corpus/fuzz"
+)
+
+func FuzzMarshalling(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		for _, ctor := range []func() any{
+			func() any { return nil },
+			func() any { return new([]any) },
+			func() any { return &OrderedMap[string, string]{} },
+			func() any { return &OrderedMap[string, any]{} },
```

Comment:
That's why you get the panic. You're supposed to use NewOrderedMap instead of creating an empty struct

Reply:
Ah but this is just a was to reproduce this usage here: https://github.com/getkin/kin-openapi/pull/695/files#diff-591476a6d1ae9c33fc8cbc4afe8f93cdcd5d4e1914034735c50c027dc67fae82R245

```go
type OAuthFlow struct {
    // ...
	Scopes           *orderedmap.OrderedMap[string, string] `json:"scopes" yaml:"scopes"`
}
```

I don't see any way of calling `New` when deserializing here. Do you?

Reply:
How about https://github.com/wk8/go-ordered-map/blob/master/json_test.go#L89-L90 ? :)

Reply:
I mean you can always do:
```
oauthFlow := &OAuthFlow{
  Scopes: orderedmap.New[string, string]()
}
json.Unmarshal(data, &oauthFlow)
```

but ok yeah, that's a bit clunky. I'm not opposed to initializing the internal map & linked list at the start of `UnmarshalJSON` if they're nil. That's fine

Reply:
Addressed in v2.1.2